src/core_functions/scalar/enum/enum_functions.cpp

```diff
@@ -28,7 +28,12 @@ static void EnumRangeFunction(DataChunk &input, ExpressionState &state, Vector &
 	for (idx_t i = 0; i < enum_size; i++) {
 		enum_values.emplace_back(enum_vector.GetValue(i));
 	}
-	auto val = Value::LIST(enum_values);
+	Value val;
+	if (!enum_values.empty()) {
+		val = Value::LIST(enum_values);
+	} else {
+		val = Value::EMPTYLIST(types[0]);
```

Comment:
`types[0]` is the enum type - the function definition states we should return a varchar list (although Postgres returns an array of the enum type instead of a varchar list, not sure if we want to do anything with that).